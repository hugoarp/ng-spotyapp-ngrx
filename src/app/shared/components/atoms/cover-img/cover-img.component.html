<figure class="cover">
  <img [src]="cover" alt="Imagen de portada" class="cover__img"/>
</figure>
