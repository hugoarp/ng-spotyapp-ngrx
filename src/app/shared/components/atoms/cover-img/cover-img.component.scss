.cover {
  &__img {
    width: 100%;
    border-radius: 0.5rem;
  }
}
