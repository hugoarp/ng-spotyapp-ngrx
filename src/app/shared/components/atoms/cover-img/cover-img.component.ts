import { Component, Input, OnInit } from '@angular/core';

@Component({
  selector: 'app-cover-img',
  templateUrl: './cover-img.component.html',
  styleUrls: ['./cover-img.component.scss'],
})
export class CoverImgComponent implements OnInit {
  @Input() cover: string = '';
  constructor() {}

  ngOnInit(): void {}
}
