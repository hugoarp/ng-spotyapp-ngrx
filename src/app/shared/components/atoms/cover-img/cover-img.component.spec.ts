import { ComponentFixture, TestBed } from '@angular/core/testing';
import { CoverImgComponent } from './cover-img.component';

describe('CoverImgComponent', () => {
  let component: CoverImgComponent;
  let fixture: ComponentFixture<CoverImgComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [CoverImgComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(CoverImgComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('debe hacer match con el snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  test('debe de tener una imagen con src y alt', () => {
    const img = compiled.querySelector('img');
    expect(img).toBeTruthy();
    expect(img?.src).toBeTruthy();
    expect(img?.alt).toBeTruthy();
  });
  
  test('debe recibir un valor de tipo string en el input cover', () => {
    const cover = 'https://picsum.photos/id/237/200/300';
    component.cover = cover;
    fixture.detectChanges();
    const img = compiled.querySelector('img');
    expect(img?.src).toBe(cover);
  })
});
