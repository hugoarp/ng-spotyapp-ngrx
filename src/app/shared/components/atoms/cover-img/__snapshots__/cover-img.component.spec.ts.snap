// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoverImgComponent debe hacer match con el snapshot 1`] = `
<div
  id="root1"
>
  <figure
    class="cover"
  >
    <img
      alt="Imagen de portada"
      class="cover__img"
      src=""
    />
  </figure>
</div>
`;
